File: Leetcode/916.cpp

class Solution {
public:
    vector<string> wordSubsets(vector<string>& A, vector<string>& B) {
        constexpr int MAXN = 26;
        int stats[MAXN]{};
        for (const auto &i : B) {
            int stats_[MAXN]{};
            for (const auto &ch : i) {
                ++stats_[ch - 'a'];
            }
            for (int j = 0; j < MAXN; ++j) {
                stats[j] = max(stats[j], stats_[j]);
            }
        }
        vector<string> result;
        bool haveAns;
        for (const auto &i : A) {
            haveAns = true;
            int stats1[MAXN]{};
            for (const auto &ch : i) {
                ++stats1[ch - 'a'];
            }
            for (int j = 0; j < MAXN; ++j) {
                if (stats1[j] < stats[j]) {
                    haveAns = false;
                    break;
                }
            }
            if (haveAns) {
                result.emplace_back(i);
            }
        }
        return result;
    }
};

File: Leetcode/916/916.go
func wordSubsets(words1 []string, words2 []string) []string {
	const N = 26
	var freq [N]int
	for i := 0; i < len(words2); i++ {
		var freq2[N] int
		for j := 0; j < len(words2[i]); j++ {
			freq2[words2[i][j] - 'a']++
		}
		for j := 0; j < N; j++ {
			if freq2[j] > freq[j] {
				freq[j] = freq2[j]
			}
		}
	}
	var result []string
	for i := 0; i < len(words1); i++ {
		var freq2[N] int
		for j := 0; j < len(words1[i]); j++ {
			freq2[words1[i][j] - 'a']++
		}
		haveAns := true
		for j := 0; j < N; j++ {
			if freq2[j] < freq[j] {
				haveAns = false
				break
			}
		}
		if haveAns {
			result = append(result, words1[i])
		}
	}
	return result
}